apple/Tests/TestCases/Profile/DeviceInfoTests.swift

```diff
@@ -76,6 +76,50 @@ final class DeviceInfoTests: Test<DeviceInfo> {
         }
         """, expected: nil)
 	}
+	
+	func test_date() {
+		
+		/// This matches `[bindings.swift.custom_types.Timestamp]`
+		/// inside `uniffi.toml` with the only difference that we use `$0` here but
+		/// the toml uses `{}`;
+		let into_custom: (String) -> Date = {let s = $0;let df = ISO8601DateFormatter();let fo = ISO8601DateFormatter.Options.withFractionalSeconds;df.formatOptions.insert(fo);func t() -> Date? {df.date(from: s)};if let d = t() {return d};df.formatOptions.remove(fo);return t()!}
+		
+		/// This matches `[bindings.swift.custom_types.Timestamp]`
+		/// inside `uniffi.toml`
+		let from_custom: (Date) -> String = { let df = ISO8601DateFormatter();df.formatOptions.insert(.withFractionalSeconds);return df.string(from: $0) }
+		
+		func testIntoThenFrom(_ vector: (String, String?)) {
+			let sut = vector.0
+			let expected = vector.1 ?? sut
+
+			let string = from_custom(into_custom(sut))
+			
+			XCTAssertEqual(string, expected)
+		}
+		
+		func testFromThenInto(_ vector: (String, String?)) {
+			let lhs = vector.0
+			let rhs = vector.1 ?? lhs
+			
+			let lhsString = from_custom(into_custom(lhs))
+			let rhsString = from_custom(into_custom(rhs))
+		
+			XCTAssertEqual(rhsString, rhs)
+			
+			XCTAssertEqual(
+				into_custom(lhsString),
+				into_custom(rhsString)
+			)
+		}
+		
+		let vectors = [
+			("2023-12-24T17:13:56.123456Z", "2023-12-24T17:13:56.123Z"), // precision lost (which is OK)
```

Comment:
Could add one where it's rounded up, to make sure it doesn't just truncate

Reply:
but it *does* truncate, this "vector" fails:

```swift
     ("2023-12-24T17:13:56.123999Z", "2023-12-24T17:13:56.124Z"),
```

but it is ok, we dont need that precision :D

Reply:
OK, but then include one that tests that it truncates I guess.

Reply:
the one above performs truncation..?

```
("2023-12-24T17:13:56.123456Z", "2023-12-24T17:13:56.123Z"),
```

please suggest an exact vector you wanna use if not that? :)